solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -5,155 +5,160 @@ import "../../../token/interfaces/ISmartToken.sol";
-        public
-        view
-        active
-        validReserve(_sourceToken)
-        validReserve(_targetToken)
-        returns (uint256, uint256)
-    {
-        // validate input
-        require(_sourceToken != _targetToken, "ERR_SAME_SOURCE_TARGET");
-
-        uint256 amount = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).crossReserveTargetAmount(
-            reserveBalance(_sourceToken),
-            reserves[_sourceToken].weight,
-            reserveBalance(_targetToken),
-            reserves[_targetToken].weight,
-            _amount
-        );
-
-        /// @dev fee is the total fee
-        uint256 fee = calculateFee(amount);
-
-        // return the amount minus the conversion fee, the conversion fee which is the total fee.
-        return (amount.sub(fee), fee);
-    }
-
-    /**
-      * @dev converts a specific amount of source tokens to target tokens
-      * can only be called by the SovrynSwap network contract
-      *
-      * @param _sourceToken source ERC20 token
-      * @param _targetToken target ERC20 token
-      * @param _amount      amount of tokens to convert (in units of the source token)
-      * @param _trader      address of the caller who executed the conversion
-      * @param _beneficiary wallet to receive the conversion result
-      *
-      * @return amount of tokens received (in units of the target token)
-    */
+	uint256 public token0Decimal;
+	uint256 public token1Decimal;
+
+	/**
+	 * @dev triggered after a conversion with new price data
+	 * deprecated, use `TokenRateUpdate` from version 28 and up
+	 *
+	 * @param  _connectorToken     reserve token
+	 * @param  _tokenSupply        smart token supply
+	 * @param  _connectorBalance   reserve balance
+	 * @param  _connectorWeight    reserve weight
+	 */
+	event PriceDataUpdate(address indexed _connectorToken, uint256 _tokenSupply, uint256 _connectorBalance, uint32 _connectorWeight);
+
+	/**
+	 * @dev initializes a new LiquidityPoolV1Converter instance
+	 *
+	 * @param  _token              pool token governed by the converter
+	 * @param  _registry           address of a contract registry contract
+	 * @param  _maxConversionFee   maximum conversion fee, represented in ppm
+	 */
+	constructor(
+		ISmartToken _token,
+		IContractRegistry _registry,
+		uint32 _maxConversionFee
+	) public LiquidityPoolConverter(_token, _registry, _maxConversionFee) {}
+
+	/**
+	 * @dev returns the converter type
+	 *
+	 * @return see the converter types in the the main contract doc
+	 */
+	function converterType() public pure returns (uint16) {
+		return uint16(CONVERTER_TYPE);
+	}
+
+	/**
+	 * @dev returns bool msg.sender is owner
+	 * use to save on contract size
+	 */
+	function isOwner() internal view returns (bool) {
+		return msg.sender == owner;
+	}
+
+	/**
+	 * @dev set oracle contract address
+	 * can be called by owner of the contract only
+	 */
+	function setOracle(address _oracle) external {
+		require(isOwner(), "ERR_SET_ORACLE_ONLY_OWNER");

```

Comment:
Changed to modifier?

Reply:
no to save contract size

Reply:
considering there is only 1 line code reverted to `onlyOwner` modifier